### PR TITLE
Hyparview: ensure removal of disconnected nodes from active/passive view

### PR DESCRIPTION
**Pull request by lrascao**
with increased test coverage

### CODE REVIEW

File: src/partisan_hyparview_peer_service_manager.erl [hunk 1]
```diff
@@ -1248,20 +1248,39 @@ prwl() ->
...
+    is_in_view(Peer, Active).
+
+-spec is_in_view(node_spec() | node(), active() | passive()) ->
+        false | {true, node_spec()}.
+is_in_view({_Node, _, _} = Peer, View) ->
+    case sets:is_element(Peer, View) of
+        false -> false;
+        true -> {true, Peer}
+    end;
+is_in_view(Node, View) ->
```
Comment (cmeiklejohn) [inline src/partisan_hyparview_peer_service_manager.erl]: Which of these views is not using the three tuple format?  That seems like a bug.

Comment (lrascao) [inline src/partisan_hyparview_peer_service_manager.erl]: the views are all correct in using the three tuple format, but lookups on it are sometimes made using this format as key, others (as occurred in the disconnect) are made using an atom node as key (which is what is being kept as key in the `Connections` dictionary)

Comment (cmeiklejohn) [inline src/partisan_hyparview_peer_service_manager.erl]: Would it make sense to remove the places where the incorrect lookups happen instead?

Comment (lrascao) [inline src/partisan_hyparview_peer_service_manager.erl]: This means refactoring the `Connections` dictionary into storing peers and not simple node atoms, i'll try it out and see how it fits with the existing code

Comment (lrascao) [inline src/partisan_hyparview_peer_service_manager.erl]: This refactor is now in progress, #55 is the first step

**Comment by lrascao** [comment]
closing in favour of #58

**Pull request closed without merging**